homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -382,6 +382,9 @@ def _extract_start_date(battery_info):
         except CarwingsError:
             _LOGGER.error("An error occurred getting battery status.")
             return None
+        except KeyError:
+            _LOGGER.error("An error occurred parsing response from server.")
```

Comment:
Please remove the periods from any log messages to follow the [style guidelines](https://developers.home-assistant.io/docs/development_guidelines/#log-messages)

```suggestion
            _LOGGER.error("An error occurred parsing response from server")
```